### PR TITLE
Fixed spelling and punctuation errors in Binary Search Trees Guide Article index.md

### PR DESCRIPTION
**Pull request by AMGB**
I have changed the spelling such as missing letters and missing punctuation and while scanning the page, I have also adjusted a few  links or coding in and out of brackets.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #XXXXX

### CODE REVIEW

**Comment by RandellDawson** [comment]
I have not reviewed all of your content, but you are misusing `it's`.  `It's` is the abbreviation of `It is`.  `Its` is the possessive form of `It`.  Please correct those first.

**Review by raisedadead — DISMISSED** [review]
Hi @AMGB 

Thanks for contributing to the codebase for first time.

From the checklist above:
> I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md)

Can you please go through this again and update the PR title to be relevant?

Thanks.

**Comment by AMGB** [comment]
> Hi @AMGB
> 
> Thanks for contributing to the codebase for first time.
> 
> From the checklist above:
> 
> > I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md)
> 
> Can you please go through this again and update the PR title to be relevant?
> 
> Thanks.

Hi,

Thanks for the feedback. What shall I change the title exactly to ?

Thanks.

**Comment by RandellDawson** [comment]
@AMGB How about something which describes what changes are being made?

Something like the following?
> Fixed spelling and Punctuation errors in Binary Search Trees Guide article

**Comment by RandellDawson** [comment]
@AMGB Thank you for making the changes.  There are about 10 PRs ahead of yours with changes to this same file.  It could be by the time we do a complete review, your PR may not be needed.  We will let you know.  If there are any merge conflicts, we will help resolve them for you.

File: guide/english/algorithms/binary-search-trees/index.md [hunk 1]
```diff
@@ -47,7 +47,7 @@ It is very similar to the search function. You again start at the root of the tr
 There are 3 cases that can happen when you are trying to delete a node. If it has,
 1. No subtree (no children): This one is the easiest one. You can simply just delete the node, without any additional actions required.
 2. One subtree (one child): You have to make sure that after the node is deleted, its child is then connected to the deleted node's parent.
-3. Two subtrees (two children): You have to find and replace the node you want to delete with its successor (the leftfmost node in the right subtree).
+3. Two subtrees (two children): You have to find and replace the node you want to delete with its successor (the left most node in the right subtree).
```
Comment (cmccormack) [inline guide/english/algorithms/binary-search-trees/index.md]: "leftmost" is the proper usage

File: guide/english/algorithms/binary-search-trees/index.md [hunk 2]
```diff
@@ -213,7 +213,7 @@ Binary search trees (BSTs) also give us quick access to predecessors and success
 ### Let's look at a couple of procedures operating on trees.
 Since trees are recursively defined, it's very common to write routines that operate on trees that are themselves recursive. 
 
-So for instance, if we want to calculate the height of a tree, that is the height of a root node, We can go ahead and recursively do that, going through the tree. So we can say:
+So for instance, if we want to calculate the height of a tree, that is the height of a root node. We can go ahead and recursively do that, going through the tree. So we can say:
```
Comment (cmccormack) [inline guide/english/algorithms/binary-search-trees/index.md]: This should not be a period, preferably "W" should be lowercase

File: guide/english/algorithms/binary-search-trees/index.md [hunk 3]
```diff
@@ -338,9 +338,9 @@ Full Binary Tree/Strict Binary Tree: A Binary Tree is full or strict if every no
         /  \        /  \
       40    50    100   40
 
-In Full Binary Tree, number of leaf nodes is equal to number of internal nodes plus one.
+In A Full Binary Tree, the number of leaf nodes is equal to number of internal nodes plus one.
```
Comment (cmccormack) [inline guide/english/algorithms/binary-search-trees/index.md]: "A" should be lowercase

File: guide/english/algorithms/binary-search-trees/index.md [hunk 4]
```diff
@@ -338,9 +338,9 @@ Full Binary Tree/Strict Binary Tree: A Binary Tree is full or strict if every no
...
  
-Complete Binary Tree: A Binary Tree is complete Binary Tree if all levels are completely filled except possibly the last level and the last level has all keys as left as possible
+Complete Binary Tree: A Binary Tree is a complete Binary Tree if all the levels are completely filled except possibly the last level and the last level has all keys as left as possible.
```
Comment (cmccormack) [inline guide/english/algorithms/binary-search-trees/index.md]: adding "the" was not necessary

File: guide/english/algorithms/binary-search-trees/index.md [hunk 5]
```diff
@@ -353,7 +353,7 @@ Complete Binary Tree: A Binary Tree is complete Binary Tree if all levels are co
...
+For example, consider a scenario where you are supposed to find the fifth smallest number in a set. You can use brute force here but we can reduce the complexity of the problem to O(lg n) by augumenting a red-black or any self-balancing tree (where n is the number of elements in the set). We can also compute rank of any element in O(lg n) time. Let us consider a case where we are augumenting a red-black tree to store the addional information needed. Besides the usual attributes, we can store a number of internal nodes in the subtree rooted at x(size of the subtree rooted at x including the node itself).
```
Comment (cmccormack) [inline guide/english/algorithms/binary-search-trees/index.md]: I believe the intention was to find the `i`th smallest number, not the fifth smallest.  `i`th is commonly used to represent any potential value.

**Review by cmccormack — DISMISSED** [review]

**Review by moT01 — APPROVED** [review]
LGTM @AMGB - congratulations on your first contribution to freeCodeCamp! 🎉

**Pull request merged**